### PR TITLE
Use markdown for checks remediation text

### PR DESCRIPTION
**Pull request by aleksei-burlakov**
This PR is for solving #50. It's only the first iteration.
This commit parses md --> html.

It was prepared during a paring programming session together with
Rubén Torrero Marijnissen <rtorreromarijnissen@suse.com>

### CODE REVIEW

**Review by arbulu89 — COMMENTED** [review]
@aleksei-burlakov It looks good!
Could you add a picture to show us how it looks like? Maybe with some real MD example in the check file.

**Comment by rtorrero** [comment]
Thanks Aleksei! It's looking good.

I made a quick test with this PR adding code-block marks in `azure-rules.yml` for the shell commands in the remediation text and it definitely looks much better though I see something weird with multi-line code blocks, which appear in a very unreadable yellow / golden color. Single line code-blocks seem to readable.

**Comment by rtorrero** [comment]
For a next iteration we need to look into making the longest remediation messages collapsed by default with maybe bootstrap collapse as @stefanotorresi suggested in issue#50

File: web/layout.go [hunk 1]
```diff
@@ -93,6 +97,19 @@ func (r *LayoutRender) addFileFromFS(templatesFS fs.FS, file string) {
 		"sum": func(a int, b int) int {
 			return a + b
 		},
+		"markup": func(md string) template.HTML {
```
Comment (stefanotorresi) [inline web/layout.go:100]: I guess the name of the function should be `markdown`? :)

File: web/layout.go [hunk 2]
```diff
@@ -93,6 +97,19 @@ func (r *LayoutRender) addFileFromFS(templatesFS fs.FS, file string) {
...
+			extensions := parser.CommonExtensions | parser.AutoHeadingIDs
+			p := parser.NewWithExtensions(extensions)
+
+			htmlFlags := html.CommonFlags | html.HrefTargetBlank
+			opts := html.RendererOptions{Flags: htmlFlags}
+			renderer := html.NewRenderer(opts)
+
+			// html := markdown.ToHTML([]byte(md), nil, nil) // could be as simple as this
```
Comment (stefanotorresi) [inline web/layout.go:108]: I don't think we need to leave this commented line behind.

File: web/templates/ha_checks.html.tmpl [hunk 3]
```diff
@@ -21,7 +21,7 @@
                     <td>{{ $Section }}</td>
                     <td>{{ .ID }}</td>
                     <td>{{ .Description }}</td>
-                    <td>{{ index .TestInfo 0 }}</td>
+                    <td>{{ markup ( index .TestInfo 0 ) }}</td>
```
Comment (stefanotorresi) [inline web/templates/ha_checks.html.tmpl:24]: see above suggestion about renaming the template function

File: web/layout.go [hunk 4]
```diff
@@ -93,6 +97,19 @@ func (r *LayoutRender) addFileFromFS(templatesFS fs.FS, file string) {
...
+			html := markdown.ToHTML([]byte(md), p, renderer)
+			result := template.HTML(html)
+			return result
```
Comment (stefanotorresi) [inline web/layout.go:111]: The `html` variable here collides with the imported `github.com/gomarkdown/markdown/html` package.
Maybe rename it as follows?
```suggestion
			rendered := markdown.ToHTML([]byte(md), p, renderer)
			result := template.HTML(rendered)
			return result
```

**Review by stefanotorresi — CHANGES_REQUESTED** [review]
Good job guys! We just need a few tweaks.

Other than the ones below, you need to update the `go.mod` and `go.sum` files after having run `go get`, otherwise the module is not declared in the project dependencies! (see the tests failing due to this)

I would also like to see a unit test of the template function; to that end, I would suggest to move the entire anonymous function to a private one in the same `layout.go` file, maybe named `func markdownToHtml(md string) template.HTML`, so that we can test it in isolation.

**Comment by stefanotorresi** [comment]
Since you folks are out the whole week I took the liberty to do the changes and merge this manually.

**Pull request closed without merging**